rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -407,35 +401,6 @@ http {
     {{ $cfg.HTTPSnippet }}
     {{ end }}
 
-    {{ if not $all.DynamicConfigurationEnabled }}
```

Comment:
Since DynamicConfiguration will be the only available option, does that mean that load-balancing algorithms other than round-robin and annotations such as:

- nginx.ingress.kubernetes.io/upstream-max-fails
- nginx.ingress.kubernetes.io/upstream-fail-timeout

will be deprecated as well. If so can they be removed as well in another commit?

Reply:
If so I can close https://github.com/kubernetes/ingress-nginx/pull/3193

Reply:
No, we can pass that info to the lua LB

Reply:
I wonder how much value they have given we "outsource" healthchecking to k8s instead of doing it in ingress-nginx controller.

I'd rather not complicate Lua balancer logic further unless it's really necessary.

Reply:
> I'd rather not complicate Lua balancer logic further unless it's really necessary.

Makes sense. That said we already have https://github.com/kubernetes/ingress-nginx/blob/master/rootfs/etc/nginx/lua/balancer.lua#L158. Maybe `upstream-max-fails` makes sense in this case?

Edit: the docs https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/balancer.md#set_more_tries are not clear about the configuration

Reply:
currently doing
```
ngx_balancer.set_more_tries(1)
```
retries the request (accring `proxy_next_upstream`) `proxy_next_upstream_tries` times. If `proxy_next_upstream_tries` is set to zero then it will try unlimited times until client timeouts or finally request succeeds.

Reply:
:+1:

Reply:
@aledbf I created https://github.com/kubernetes/ingress-nginx/pull/3207 to do so